inspectit-oce-core/src/main/java/rocks/inspectit/oce/core/instrumentation/NewClassDiscoveryService.java

```diff
@@ -0,0 +1,79 @@
+package rocks.inspectit.oce.core.instrumentation;
+
+import lombok.extern.slf4j.Slf4j;
+import lombok.val;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import rocks.inspectit.oce.core.config.InspectitEnvironment;
+import rocks.inspectit.oce.core.utils.StopWatch;
+
+import javax.annotation.PostConstruct;
+import java.lang.instrument.Instrumentation;
+import java.time.Duration;
+import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+
+@Service
+@Slf4j
+public class NewClassDiscoveryService {
```

Comment:
unit test as well.. make sure to simulate new class loading in between new checks..